meteor/client/lib/forms/schemaFormWithOverrides.tsx

```diff
@@ -0,0 +1,281 @@
+import { IntInputControl } from '../Components/IntInput'
+import { JsonTextInputControl } from '../Components/JsonTextInput'
+import {
+	LabelAndOverrides,
+	LabelAndOverridesForCheckbox,
+	LabelAndOverridesForDropdown,
+	LabelAndOverridesForInt,
+} from '../Components/LabelAndOverrides'
+import { MultiLineTextInputControl } from '../Components/MultiLineTextInput'
+import { TextInputControl } from '../Components/TextInput'
+import { JSONSchema, TypeName } from '@sofie-automation/shared-lib/dist/lib/JSONSchemaTypes'
+import { SchemaFormTable } from './schemaFormTable'
+import { joinObjectPathFragments, translateStringIfHasNamespaces } from './schemaFormUtil'
+
+export interface SchemaFormCommonProps {
+	schema: JSONSchema
+	translationNamespaces: string[]
+
+	/**
+	 * In some situations, it is desirable to not allow tables to be used by a schema
+	 * For example, a table inside a table will not display properly so this gets set automatically
+	 */
+	allowTables?: boolean
+}
+
+interface SchemaFormWithOverridesProps extends SchemaFormCommonProps {
+	attr: string
+
+	item: WrappedOverridableItemNormal<any>
+	overrideHelper: OverrideOpHelper
+}
+
+interface FormComponentProps {
+	schema: JSONSchema
+	translationNamespaces: string[]
+	commonAttrs: {
+		label: string
+		hint: string | undefined
+
+		item: WrappedOverridableItemNormal<any>
+		overrideHelper: OverrideOpHelper
+		itemKey: string
+		opPrefix: string
+	}
+}
+
+function useChildPropsForFormComponent(props: SchemaFormWithOverridesProps) {
+	return useMemo(() => {
+		const title = props.schema['ui:title'] || props.attr
+		const description = props.schema['ui:description']
+
+		return {
+			schema: props.schema,
+			translationNamespaces: props.translationNamespaces,
+			commonAttrs: {
+				label: translateStringIfHasNamespaces(title, props.translationNamespaces),
+				hint: description ? translateStringIfHasNamespaces(description, props.translationNamespaces) : undefined,
+				item: props.item,
+				itemKey: props.attr,
+				opPrefix: props.item.id,
+				overrideHelper: props.overrideHelper,
+			},
+		}
+	}, [props.schema, props.translationNamespaces, props.attr, props.item, props.overrideHelper])
+}
+
+export function SchemaFormWithOverrides(props: SchemaFormWithOverridesProps) {
+	const { t } = useTranslation()
+
+	const childProps = useChildPropsForFormComponent(props)
+
+	switch (props.schema.type) {
+		case TypeName.Array:
+			return <ArrayFormWithOverrides {...props} />
+		case TypeName.Object:
+			if (props.schema['ui:displayType'] === 'json') {
+				return <JsonFormWithOverrides {...childProps} />
+			} else {
+				return <ObjectFormWithOverrides {...props} />
+			}
+		case TypeName.Integer:
+			if (props.schema.enum) {
+				return <EnumFormWithOverrides {...childProps} />
+			} else {
+				return <IntegerFormWithOverrides {...childProps} />
+			}
+		case TypeName.Number:
+			return <NumberFormWithOverrides {...childProps} />
```

Comment:
I'd like these to respect JSON Schema `multipleOf`, `minimum` and `maximum`, and pass them down to the HTML input elements, but I guess it's not necessarily an MVP thing, since the Config Manifest system did not have that capability either.

Reply:
I have implemented minimum and maximum.
I think `multipleOf` requires some more implementation as its doesn't quite line up with the behaviour of step, so shall leave that for later.